odl/tomo/geometry/detector.py

```diff
@@ -0,0 +1,459 @@
+        """
+
+    @property
+    def params(self):
+        """Surface parameter set of this geometry."""
+        return self._params
+
+    @property
+    def param_grid(self):
+        """The sampling grid for the parameters."""
+        return self._param_grid
+
+    @property
+    def has_sampling(self):
+        """Return `True` if a sampling grid is given, else `False`."""
+        return self.param_grid is not None
+
+    @property
+    def npixels(self):
+        """The number of pixels (sampling points)."""
+        if not self.has_sampling:
+            raise ValueError('no sampling defined for {}.'.format(self))
+        return self.param_grid.shape
+
+    def surface_deriv(self, param):
+        """The partial derivative(s) of the surface parametrization.
+
+        Parameters
+        ----------
+        param : element of `params`
+            The parameter value where to evaluate the function
+        """
+        raise NotImplementedError
+
+    def surface_measure(self, param):
+        """The density function of the surface measure.
+
+        This is the default implementation relying on the `surface_deriv`
+        method. For ``ndim == 1``, the density is given by the `Arc
+        length`_, for ``ndim == 2``, it is the length of the cross product
+        of the partial derivatives of the parametrization, see Wikipedia's
+        `Surface area`_ article.
+
+        Parameters
+        ----------
+        param : element of `params`
+            The parameter value where to evaluate the function
+
+        Returns
+        -------
+        measure : `float`
+            The density value at the given parameter
+
+        .. _Arc length:
+            https://en.wikipedia.org/wiki/Curve#Lengths_of_curves
+        .. _Surface area:
+            https://en.wikipedia.org/wiki/Surface_area
+        """
+        if param not in self.params:
+            raise ValueError('parameter value {} not in the valid range {}.'
+                             ''.format(param, self.params))
+        if self.ndim == 1:
+            return float(np.linalg.norm(self.surface_deriv(param)))
+        elif self.ndim == 2:
+            return float(np.linalg.norm(np.cross(*self.surface_deriv(param))))
+        else:
+            raise NotImplementedError
+
+
+class FlatDetector(with_metaclass(ABCMeta, Detector)):
+
+    """Abstract class for flat detectors in 2 and 3 dimensions."""
+
+    def surface_measure(self, param):
+        """The constant density function of the surface measure.
+
+        Parameters
+        ----------
+        param : element of `params`
+            The parameter value where to evaluate the function
+
+        Returns
+        -------
+        meas : `float`
+            The constant density 1.0
+        """
+        if param not in self.params:
+            raise ValueError('parameter value {} not in the valid range '
+                             '{}.'.format(param, self.params))
+        return 1.0
```

Comment:
That is handled in the underlying discretization. This is the weighting function accounting for non-flat detectors.